test/junit/scala/lang/stringinterpol/StringContextTest.scala

```diff
@@ -252,4 +252,103 @@ class StringContextTest {
+      (StringContext.glob(Seq("abc"), "aabc") -> None),
+      (StringContext.glob(Seq("abc"), "abcb") -> None),
+      (StringContext.glob(Seq("abc"), "cba")  -> None),
+      (StringContext.glob(Seq("abc"), "abbc") -> None),
+  
+      // Single wildcard
+      (StringContext.glob(Seq("a", "b"), "ab")      -> Some(Seq(""))),
+      (StringContext.glob(Seq("a", "b"), "a b")     -> Some(Seq(" "))),
+      (StringContext.glob(Seq("a", "b"), "a    b")  -> Some(Seq("    "))),
+      (StringContext.glob(Seq("a", "b"), "a***b")   -> Some(Seq("***"))),
+      (StringContext.glob(Seq("a", "b"), "a")       -> None),
+      (StringContext.glob(Seq("a", "b"), "b")       -> None),
+      (StringContext.glob(Seq("a", "b"), "abc")     -> None),
+      (StringContext.glob(Seq("a", "b"), "cab")     -> None),
+      (StringContext.glob(Seq("a", "b"), " ab ")    -> None),
+  
+      // Leading wildcard
+      (StringContext.glob(Seq("", "b"), "ab")     -> Some(Seq("a"))),
+      (StringContext.glob(Seq("", "b"), "a b")    -> Some(Seq("a "))),
+      (StringContext.glob(Seq("", "b"), "a    b") -> Some(Seq("a    "))),
+      (StringContext.glob(Seq("", "b"), "a***b")  -> Some(Seq("a***"))),
+      (StringContext.glob(Seq("", "b"), "a")      -> None),
+      (StringContext.glob(Seq("", "b"), "b")      -> Some(Seq(""))),
+      (StringContext.glob(Seq("", "b"), "abc")    -> None),
+      (StringContext.glob(Seq("", "b"), "cab")    -> Some(Seq("ca"))),
+      (StringContext.glob(Seq("", "b"), " ab ")   -> None),
+  
+      // Trailing wildcard
+      (StringContext.glob(Seq("a", ""), "ab")     -> Some(Seq("b"))),
+      (StringContext.glob(Seq("a", ""), "a b")    -> Some(Seq(" b"))),
+      (StringContext.glob(Seq("a", ""), "a    b") -> Some(Seq("    b"))),
+      (StringContext.glob(Seq("a", ""), "a***b")  -> Some(Seq("***b"))),
+      (StringContext.glob(Seq("a", ""), "a")      -> Some(Seq(""))),
+      (StringContext.glob(Seq("a", ""), "b")      -> None),
+      (StringContext.glob(Seq("a", ""), "abc")    -> Some(Seq("bc"))),
+      (StringContext.glob(Seq("a", ""), "cab")    -> None),
+      (StringContext.glob(Seq("a", ""), " ab ")   -> None),
+  
+      // Lonely wildcard
+      (StringContext.glob(Seq("", ""), "ab")      -> Some(Seq("ab"))),
+      (StringContext.glob(Seq("", ""), "a b")     -> Some(Seq("a b"))),
+      (StringContext.glob(Seq("", ""), "a    b")  -> Some(Seq("a    b"))),
+      (StringContext.glob(Seq("", ""), "a***b")   -> Some(Seq("a***b"))),
+      (StringContext.glob(Seq("", ""), "a")       -> Some(Seq("a"))),
+      (StringContext.glob(Seq("", ""), "b")       -> Some(Seq("b"))),
+      (StringContext.glob(Seq("", ""), "abc")     -> Some(Seq("abc"))),
+      (StringContext.glob(Seq("", ""), "cab")     -> Some(Seq("cab"))),
+      (StringContext.glob(Seq("", ""), " ab ")    -> Some(Seq(" ab "))),
+  
+      // Ambiguous lonely wildcard
+      (StringContext.glob(Seq("", "", ""), "ab")      -> Some(Seq("", "ab"))),
```

Comment:
Maybe document that stars are reluctant: `val s"${a}x${b}" = "axxxb"` gives `a`, `xxb` instead of `axx`. Maybe add that unit test sample, since I think in the covering case, I can't tell which `b` is shown. Or no, L338 is a,b not ab,empty.